state-sync/storage-service/server/src/lib.rs

```diff
@@ -525,40 +525,42 @@ impl StorageReader {
     }
 
     /// Returns the transaction range held in the database (lowest to highest).
-    fn fetch_transaction_range(
+    fn fetch_ledger_range(
         &self,
         latest_version: Version,
     ) -> Result<Option<CompleteDataRange<Version>>, Error> {
-        let first_transaction_version = self
+        let pruning_window_option = self
             .storage
-            .get_first_txn_version()
-            .map_err(|error| Error::StorageErrorEncountered(error.to_string()))?;
-        if let Some(first_transaction_version) = first_transaction_version {
-            let transaction_range =
-                CompleteDataRange::new(first_transaction_version, latest_version)
-                    .map_err(|error| Error::UnexpectedErrorEncountered(error.to_string()))?;
-            Ok(Some(transaction_range))
-        } else {
-            Ok(None)
-        }
-    }
-
-    /// Returns the transaction output range held in the database (lowest to highest).
-    fn fetch_transaction_output_range(
-        &self,
-        latest_version: Version,
-    ) -> Result<Option<CompleteDataRange<Version>>, Error> {
-        let first_output_version = self
-            .storage
-            .get_first_write_set_version()
-            .map_err(|error| Error::StorageErrorEncountered(error.to_string()))?;
-        if let Some(first_output_version) = first_output_version {
-            let output_range = CompleteDataRange::new(first_output_version, latest_version)
-                .map_err(|error| Error::UnexpectedErrorEncountered(error.to_string()))?;
-            Ok(Some(output_range))
+            .get_ledger_prune_window()
+            .map_err(|error| Error::StorageErrorEncountered(error.to_string()))?
+            .map(|window| window as u64);
+        let lowest_transaction_version = if let Some(pruning_window) = pruning_window_option {
+            if latest_version < pruning_window {
+                // No pruning has occured, we can safely assume that the first transaction version is 0
+                0
+            } else {
+                // lowest_account_version = latest_version - pruning_window + 1;
```

Comment:
The logic is not sound because one can enlarge the pruning window between runs...

@JoshLind I known in practice the lowest version isn't really useful, can live with this?

Reply:
Yeah, we also have the same issue if someone has the pruner enabled for a while, then restarts with it disabled. I'm in two minds about this change.
- I'm less worried about account states because we'll always try to get the latest by default. But, with transactions/outputs, I think there's a valid reason for wanting to run a node that fetches everything.
- How hard/ugly is it to maintain this API and try to keep it accurate/correct? Is it performance we're worried about or something else?

Reply:
@JoshLind - Its a performance concern - especially because the API to get first txn version needs to seek to the first txn - before pruning enabled this was cheap because we can cheaply seek to the first txn version but after pruning that is not the case anymore. 

>> But, with transactions/outputs, I think there's a valid reason for wanting to run a node that fetches everything.

If we want to run a node that fetches everything, we can just turn off pruning - and this API should behave as expected.

Reply:
> before pruning enabled this was cheap because we can cheaply seek to the first txn version but after pruning that is not the case anymore.

It just came to me that now that we don't use `range_delete`s, performance might now be a concern any more.

Reply:
It's still more expensive than before right? Because we would need to skip a bunch of deleted entries in the DB?

Reply:
hmm, now I'm not sure. @lightmark  insights?

My hope is with out the range_deletes, this is as fast or slow as a normal GET. So the worst case we can cache it.

Reply:
Alright, I added this back for now, I can plan to run some more tests to understand if this is still an issue without range_deletes. If so, we can address this in a separate PR either via current approach or via caching it.

Reply:
👌